yarn-project/sequencer-client/src/client/sequencer-client.ts

```diff
@@ -64,20 +60,14 @@ export class SequencerClient {
     contractDataSource: ContractDataSource,
     l2BlockSource: L2BlockSource,
     l1ToL2MessageSource: L1ToL2MessageSource,
+    prover: ProverClient,
```

Comment:
I think this should be `BlockProver`. Otherwise it becomes unclear who "owns" the `ProverClient`: the Sequencer(Client) or the Node.